draco/synthesis/mockcatalog.py

```diff
@@ -318,8 +318,16 @@ class PdfGeneratorBase(task.SingleTask):
     by the task :class:`MockCatalogGenerator` to draw mock catalogs.
 
     Derived classes must implement process().
+
+    Attributes
+    ----------
+    tracer : str, optional
```

Comment:
It would also make sense to have a `tracer` parameter in `SelectionFunctionEstimator`, and have it stored as an attribute in the selection function map and propagated forward from there. It is also nice to have the `tracer` parameter where you've put it, though, in case the user wants to specify the tracer but isn't using a selection function. Perhaps we could have both options, and add code to `PdfGeneratorWithSelectionFunction` to check for a tracer attribute in the selection function container if none is specified in `PdfGeneratorWithSelectionFunction` itself. What do you think?

Reply:
Good point. I guess maybe the thing to do is to move the functionality onto `draco.core.task.SingleTask` to be a generic way of setting an attribute on the output of any task.

Reply:
Okay. I've added the same code into the SelectionFunctionEstimator. I'll make an issue for the generic functionality.